### ISSUE TITLE
Add function in Post2D window to visualize calculation results defined at cell center

### ISSUE THREAD

**Issue by kskinoue0612** [i-RIC/prepost-gui#154]
## Abstract

iRIClib now has function to export calculation result defined at cell center, like `cg_iric_write_sol_cell_integer_f`, `cg_iric_write_sol_cell_real_f`, but iRIC GUI do not have function to visualize the result exported with these functions.

So, we add functions to visualize the results defined at cell centers.

## Post2D window

Add function to show "colored cells" in Post2D window, for visualizing scalar values.

### Object Browser

Add "Scalar (Cell Center)" in the object browser. Not sure where is the best position, to add, but maybe at the bottom is a option

### How to visualize

The value is the same in the whole area of cells, so each cells are filled with the same color for each, as the image below.

![post2d](https://user-images.githubusercontent.com/4031569/29904516-296d68c2-8e44-11e7-8460-b6f623c7f82c.png)

### The dialog to setup

Basically it is similar to "Scalar" dialog used for result defined at grid nodes, that is displayed below.

![dialog](https://user-images.githubusercontent.com/4031569/29904522-320aa634-8e44-11e7-934d-8b8ed2c558aa.png)

The difference will be:
* "Isolines" will be removed from "Display Setting".

## Bird's eye Post2D window

We add the same function described above, to Bird's eye Post2D window too.

## Notice

I think we should add functions to visualize "Arrows", "Streamlines", "Particles" using calculation results defined at grid nodes, but they will be less important.

So, let's add this function first, and go forward to implement those functions, when this is finished.

**Comment by kskinoue0612** [comment]
Hi Scott

This is the function that I want you to implement, when https://github.com/i-RIC/iriclib/issues/15 is closed.

The description above is just a draft idea, and I welcome comments or proposal of improvements on it. Please add comments, or edit the description above directly.

Thanks,
Keisuke

**Comment by scharlton2** [comment]
[Braid-Cell.zip](https://github.com/i-RIC/prepost-gui/files/1381431/Braid-Cell.zip)

**Comment by kskinoue0612** [comment]
Hi Scott, 

Thank you for sending the file.
I also think that it will be a good way to pass the position as an argument, to Post2dWindowNodeScalarGroupDataItem::setupColorFringeSetting(). I have no better idea .

Thanks,
Keisuke

**Comment by scharlton2** [comment]
Hi Keisuke,

I wasn't able to find the file I thought I had uploaded, so I re-uploaded it here:
[Braid-Cell-8.zip](https://github.com/i-RIC/prepost-gui/files/1448670/Braid-Cell-8.zip)

And here's a file to test the unstructured grid:
[CellCenterProject.4.zip](https://github.com/i-RIC/prepost-gui/files/1448679/CellCenterProject.4.zip)

Thanks,
Scott

**Comment by scharlton2** [comment]
Hi Keisuke,

You might need to apply this patch (something you recently fixed) in order for the examples to work.


```
diff --git a/libs/geodata/polygon/geodatapolygon.cpp b/libs/geodata/polygon/geodatapolygon.cpp
index e5dad992..491b5913 100644
--- a/libs/geodata/polygon/geodatapolygon.cpp
+++ b/libs/geodata/polygon/geodatapolygon.cpp
@@ -934,8 +934,12 @@ void GeoDataPolygon::initParams()
                maxIndex = dimensions()->maxIndex();
        }
        bool ok;
-       double defaultValue = gridAttribute()->variantDefaultValue().toDouble(&ok);
-       if (! ok) {defaultValue = 0;}
+       double defaultValue = 0; 
+       auto gridAtt = gridAttribute();
+       if (gridAtt != nullptr) {
+               defaultValue = gridAtt->variantDefaultValue().toDouble(&ok);
+               if (!ok) { defaultValue = 0; }
+       }

        for (int i = 0; i <= maxIndex; ++i) {
                m_variantValues.push_back(defaultValue);

```

Thanks,
Scott
